File: src/main/java/dev/robothanzo/werewolf/database/documents/UserRole.java
package dev.robothanzo.werewolf.database.documents;

import lombok.Getter;

@Getter
public enum UserRole {
    JUDGE("法官"),
    SPECTATOR("觀眾"),
    PENDING("待定"),
    BLOCKED("已封鎖");

    private final String description;

    UserRole(String description) {
        this.description = description;
    }

    public boolean isPrivileged() {
        return this == JUDGE || this == SPECTATOR;
    }

    public static UserRole fromString(String role) {
        if (role == null)
            return PENDING;
        try {
            return UserRole.valueOf(role.toUpperCase());
        } catch (IllegalArgumentException e) {
            return PENDING;
        }
    }
}

File: src/main/kotlin/dev/robothanzo/werewolf/database/documents/UserRole.kt
package dev.robothanzo.werewolf.database.documents

enum class UserRole(val description: String) {
    JUDGE("法官"),
    SPECTATOR("觀眾"),
    PENDING("待定"),
    BLOCKED("已封鎖");

    fun isPrivileged(): Boolean {
        return this == JUDGE || this == SPECTATOR
    }

    companion object {
        fun fromString(role: String?): UserRole {
            if (role == null) return PENDING
            return try {
                valueOf(role.uppercase())
            } catch (e: IllegalArgumentException) {
                PENDING
            }
        }
    }
}
